src/Controller/StripeWebhookEndpoint.php

```diff
@@ -244,19 +258,20 @@ private function onAccountUpdated(Event $event): string
      * @throws \Doctrine\ORM\ORMException
      * @throws \Doctrine\ORM\OptimisticLockException
      */
-    private function onPaymentIntentOrChargeCreated(Event $event): string
+    private function onChargeCreated(Event $event): string
     {
         $apiStripePayment = $event->data->object;
+        if (!$apiStripePayment instanceof \Stripe\Charge) {
+            $message = sprintf('Webhook expected a charge. Received a %s instead.', \get_class($apiStripePayment));
```

Comment:
The type is under the `object` attribute of the message. Both work but it might be safer to compare a string than a class, e.g. `$apiStripePayment['object'] === 'charge'`.

Reply:
By comparing the class PHP (and PHPStan) infers the type of the object.
I can use the fully types a `\Stripe\Charge` and not the `Stripe\StripeObject`.

Reply:
Good point!